codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -106,8 +107,8 @@ public Configuration configuration() {
     }
 
     @Override
-    public Http2HeaderTable headerTable() {
-        return headersEncoder.configuration().headerTable();
+    public Http2HeadersEncoder.Configuration headersConfiguration() {
```

Comment:
The rational is we don't want to expose the `Http2HeadersEncoder` which may make it easier for the user to call `Http2HeadersEncoder#encodeHeaders` which will result in HPACK getting out of sync. Did you have an alternative approach in mind?